transaction/std/src/test_utils.rs

```diff
@@ -0,0 +1,206 @@
+//! Utilities that help with testing the transaction builder and related objects
+
+use crate::{EmptyMemoBuilder, InputCredentials, MemoPayload, TransactionBuilder, TxBuilderError};
+use core::convert::TryFrom;
+use mc_account_keys::{AccountKey, PublicAddress, DEFAULT_SUBADDRESS_INDEX};
+use mc_crypto_keys::RistrettoPublic;
+use mc_fog_report_validation_test_utils::FogPubkeyResolver;
+use mc_transaction_core::{
+    onetime_keys::*,
+    tokens::Mob,
+    tx::{Tx, TxOut, TxOutMembershipProof},
+    Amount, BlockVersion, Token, TokenId,
+};
+use rand::{CryptoRng, RngCore};
+
+/// Creates a TxOut that sends `value` to `recipient`.
+///
+/// Note: This is only used in test code
+///
+/// # Arguments
+/// * `value` - Value of the output, in picoMOB.
+/// * `recipient` - Recipient's address.
+/// * `fog_resolver` - Set of prefetched fog public keys to choose from
+/// * `rng` - Entropy for the encryption.
+///
+/// # Returns
+/// * A transaction output, and the shared secret for this TxOut.
+pub fn create_output<RNG: CryptoRng + RngCore, FPR: FogPubkeyResolver>(
+    block_version: BlockVersion,
+    amount: Amount,
+    recipient: &PublicAddress,
+    fog_resolver: &FPR,
+    rng: &mut RNG,
+) -> Result<(TxOut, RistrettoPublic), TxBuilderError> {
+    let (hint, _pubkey_expiry) =
+        crate::transaction_builder::create_fog_hint(recipient, fog_resolver, rng)?;
+    crate::transaction_builder::create_output_with_fog_hint(
+        block_version,
+        amount,
+        recipient,
+        hint,
+        |_| {
+            Ok(if block_version.e_memo_feature_is_supported() {
```

Comment:
Consider https://doc.rust-lang.org/std/primitive.bool.html#method.then
I have never used it but if it is available I think it is a bit more elegant

Reply:
cool will try it